File: be/src/main/java/com/netcracker/edu/backend/entity/BillingAccount.java
package com.netcracker.edu.backend.entity;

import javax.persistence.*;
import java.util.Objects;

@Entity
@Table(name = "billing_account")
public class BillingAccount {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    private String address;
    private String username;
    private String email;

    public BillingAccount(String address, String username, String email) {
        this.address = address;
        this.username = username;
        this.email = email;
    }

    public BillingAccount() {

    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        BillingAccount that = (BillingAccount) o;
        return id == that.id &&
                Objects.equals(address, that.address) &&
                Objects.equals(username, that.username) &&
                Objects.equals(email, that.email);
    }

    @Override
    public int hashCode() {

        return Objects.hash(id, address, username, email);
    }

    @Override
    public String toString() {
        return "BillingAccount{" +
            "id=" + id +
            ", address='" + address + '\'' +
            ", username='" + username + '\'' +
            ", email='" + email + '\'' +
            '}';
    }
}

File: frontend/src/app/modules/components/billing-account/models/billing-account.ts
export class BillingAccount {
  id: string;
  address: string;
  username: string;
  email: string;

  static cloneBase(billingAccout: BillingAccount): BillingAccount {
    const clonedBillingAccount: BillingAccount = new BillingAccount();
    clonedBillingAccount.id = billingAccout.id;
    clonedBillingAccount.username = billingAccout.username;
    clonedBillingAccount.email = billingAccout.email;
    clonedBillingAccount.address = billingAccout.address;
    return clonedBillingAccount;
  }
}

export class BillingAccountStr {
}
